Fix all tests.

Ismael-VC opened the pull request:
Make tast pass so that the package can be loaded again.

* [x] Fix StreamToken constructor call, see https://github.com/jakebolewski/JuliaParser.jl/pull/31 & https://github.com/jakebolewski/JuliaParser.jl/issues/29
* [x] Fix utf8sizeof import.
* [x] Fix testfile abs path.
* [x] Fix line_number_node, see https://github.com/MichaelHatherly/Lexicon.jl/issues/131 & https://github.com/MichaelHatherly/Docile.jl/commit/a2509fb62c58409ae8382cb25d7363a2d75118b3

Ismael-VC commented:
oh damn! how can I reopen this @jakebolewski I'm sorry don't know what happended, but anyway, this lets me load the Package agian.

Ismael-VC commented:
Should I open another PR? Sorry for the noise! :/

Ismael-VC commented:
Could we merge this?, by the tests, you can see that the this PR makes the package loadable again, the other errors need their own issues IMHO.

tkelman commented on test/parser.jl:
```diff
@@ -24,7 +24,7 @@ facts("test parse IOBuffer") do
 
 facts("test parse IOStream") do
-    io = open("testfile")
+    io = open(joinpath(Pkg.dir("JuliaParser"), "test", "testfile"))
```
this should use `dirname(@__FILE__)`, not `Pkg.dir`

Ismael-VC replied:
Got it!

Ismael-VC replied:
@tkelman fixed and squased, thanks!

Ismael-VC commented:
I fixed the PR as per Tony suggestion, and squashed. Again this makes the module loadable again, failing CI is out of scope for this PR. I think it's ready for merge.

tkelman commented on src/parser.jl:
```diff
@@ -130,7 +130,12 @@ end
 
-line_number_node(ts) = LineNumberNode(curline(ts))
+if VERSION < v"0.4-dev+7034"
+    line_number_node(f, n) = LineNumberNode(n)
```
why are you defining `line_number_node` with 2 arguments when it only gets used with one input?

Ismael-VC replied:
It's in the desctiption, see: https://github.com/MichaelHatherly/Lexicon.jl/issues/131 & https://github.com/MichaelHatherly/Docile.jl/commit/a2509fb62c58409ae8382cb25d7363a2d75118b3

Ismael-VC replied:
Specifically here: https://github.com/MichaelHatherly/Docile.jl/commit/a2509fb62c58409ae8382cb25d7363a2d75118b3#diff-0d3c997012e03e7eb1c62f1e8c4e712eR200

tkelman replied:
How is the fix Lexicon/Docile needed relevant here? This function may have the same name but I don't think it's used the same way here.

Ismael-VC replied:
When fixing the other errors I eventually got the same error that Lexicon had: `ERROR: LoadError: LoadError: MethodError: `convert` has no method matching convert(::Type{LineNumberNode}, ::Int64)
` I searched for it in Google, found Docile commit, tried that and it fixed the issue here.

tkelman replied:
how does defining an unused method help in any way?

Ismael-VC replied:
You are right only `line_number_node(ts) = LineNumberNode(symbol(""), curline(ts))` is needed. And indeed the Package is loadable but not able to parse anything, what this PR does (I will update of course) is make `JuliaParser.Parser.parse` work again also, I think I choose bad names for the PR.

tkelman replied:
see the 0.3 test log on travis. missing method.

Ismael-VC replied:
I'll check it out, actually it should be something like:

```julia
if VERSION < v"0.4-dev+7034"
    line_number_node(ts) = LineNumberNode(curline(ts))
else
    line_number_node(ts) = LineNumberNode(:none, curline(ts))
end

Keno replied:
Yes, this looks right.

tkelman commented:
I think master is loadable now. I don't know whether this is correct but if you can get tests passing I'll merge.

Ismael-VC commented:
You mean all the tests? There are several different errors, I think they need their own issues and PRs, but I'll give it a try.

tkelman commented:
Unless someone who actually knows all the code here shows up and reviews this change, I don't have any other way of judging correctness other than tests passing vs failing. At some point in the past I'm assuming the package passed tests, though I don't know how long ago that was.

Ismael-VC commented:
I'll just try to fix everything.

Keno commented:
I'll merge this with the change you suggested for line_number_node.

Keno commented:
After this change the tests are failing due to https://github.com/JuliaLang/julia/issues/14658. I think I'll actually hold off until that's resolved.

Keno commented:
After applying the base julia workaround and this, most tests work again, so I've merged this in 11479456ca7a0c8f92de33ad01271a12e6426fa6. I'll work through the remaining tests.

Ismael-VC commented:
Thanks @Keno, https://github.com/JuliaLang/JuliaParser.jl/commit/11479456ca7a0c8f92de33ad01271a12e6426fa6 also closes #29 and #31.
